app/controllers/web-payments/apple-pay/normalise-apple-pay-payload.js

```diff
@@ -80,7 +81,14 @@ const nullable = word => {
   return word
 }
 
-const normaliseLastDigitsCardNumber = displayName => displayName.substr(displayName.length - 4)
+const normaliseLastDigitsCardNumber = displayName => {
+  let lastDigitsCardNumber = ''
+  const lastFourChars = displayName.substr(displayName.length - 4)
```

Comment:
You can actually pass a negative number to `substr(…)` to have it start counting from the end of the string rather than the beginning, so `displayName.substr(-4)` would be equivalent if `displayName` is less than 4 characters long.

I think this be easier to understand if the substring was taken after the `displayName.length >= 4` check, so the reader doesn’t have to try to figure out what  `displayName.substr(-4)` will do if `displayName` is under four characters long.

Reply:
We are going do the Pact tests in another PR.